hanadb_exporter/exporters/prometheus_exporter.py

```diff
@@ -31,48 +32,63 @@ class SapHanaCollector(object):
     SAP HANA database data exporter
     """
 
-    def __init__(self, conector):
-        super(SapHanaCollector, self).__init__()
+    def __init__(self, connector):
         self._logger = logging.getLogger(__name__)
-        self._hdb_connector = conector
+        self._hdb_connector = connector
 
-    def _execute(self, metric):
+
+    def _format_query_result(self, query_result):
         """
-        Create metric object
+        Format query results to match column names with their values for each row
+        Returns nested list, containing tuples (column_name, value)
 
         Args:
-            metric (dict): query, info, type structure dictionary
+            query_result (obj): QueryResult object
         """
-        try:
-            value = self._hdb_connector.query(metric['query'])
-
-            if metric['type'] == core.GaugeMetricFamily:
-                metric_obj = self._manage_gauge(metric, value)
-            else:
-                raise NotImplementedError('{} type not implemented'.format(metric['type']))
-
-            return metric_obj
-        except KeyError as err:
-            raise MalformedMetric(err)
-
-    def _manage_gauge(self, metric, value):
+        query_columns = []
+        formatted_query_result = []
+        query_columns = [meta[0] for meta in query_result.metadata]
+        for record in query_result.records:
+            formatted_query_result.append(list(itertools.izip(query_columns, record)))
```

Comment:
```suggestion
            formatted_query_result.append(list(zip(query_columns, record)))
```
This does not work on Python3 as izip was moved as `zip()` into the standard library. The import can be removed too.
My POV is that we should use py3 only for the development as it is a new project and include CI checks for that.

Reply:
@diegoakechi @ayoub-belarbi Besides that, I have other question. Can we have more than one record in our queries?
As far as I understand, they should only return one record, right?
If this was the case, we could remove the `for` loop too

Reply:
And to fix the issue with `zip` we could update the import to look like:
```
try:
    import itertools.izip as zip
except ImportError:
    pass
```
https://python-future.org/compatible_idioms.html

Reply:
@arbulu89 Yes, we can. For example, the CPU query returns one line for each CPU Core, so the loop is needed to avoid unnecessary SQL executions to the database, as all the information can be retrieved in one shot.